corehq/apps/domain/decorators.py

```diff
@@ -225,6 +234,30 @@ def _inner(request, *args, **kwargs):
     return _inner
 
 
+def check_lockout(fn):
+    @wraps(fn)
+    def _inner(request, *args, **kwargs):
+        username = _get_username_from_request(request)
+        user = CouchUser.get_by_username(username)
```

Comment:
If it is preferable, I can add the couch user to the kwargs of the wrapped function and then check for that in the other `_inner` function above before doing the database hit. It feels slightly wrong to mutate the kwargs that are supposed to be passed through to the view function and other auth decorators but if we pop that value as soon as we use it and are careful to give it a unique name, it could save the database hit. It would only affect calls through the API but it would be every API hit.